### PR TITLE
fix(platform-server): allow multiple instances of platformServer and platformDynamicServer

### PR DESCRIPTION
**Pull request by vikerman**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
There can be only one instance of PlatformRef. But we need multiple instances of platformServer and platformDynamicServer

**What is the new behavior?**
allow multiple instances of platformServer and platformDynamicServer

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: modules/@angular/core/src/application_ref.ts [hunk 1]
```diff
@@ -83,9 +84,11 @@ export class NgProbeToken {
...
-  if (_platform && !_platform.destroyed) {
-    throw new Error(
-        'There can be only one platform. Destroy the previous one to create a new one.');
+  if (!_allowMultiplePlatforms) {
+    if (_platform && !_platform.destroyed) {
+      throw new Error(
+          'There can be only one platform. Destroy the previous one to create a new one.');
+    }
   }
   _platform = injector.get(PlatformRef);
```
Comment (alxhub) [inline modules/@angular/core/src/application_ref.ts]: Let's not set `_platform` in the multiple-platform case, to make absolutely sure there isn't a chance of cross-talk between platforms.

**Review by alxhub — APPROVED** [review]
LGTM on the platform changes, but I assume the other commits in this PR are separately reviewed

**Comment by vikerman** [comment]
#13822

File: modules/@angular/core/src/application_ref.ts [hunk 2]
```diff
@@ -103,12 +106,13 @@ export function createPlatformFactory(
     providers: Provider[] = []): (extraProviders?: Provider[]) => PlatformRef {
   const marker = new InjectionToken(`Platform: ${name}`);
   return (extraProviders: Provider[] = []) => {
-    if (!getPlatform()) {
+    let platform = getPlatform();
+    if (!platform || platform.injector.get(ALLOW_MULTIPLE_PLATFORMS, false)) {
       if (parentPlatformFactory) {
-        parentPlatformFactory(
+        platform = parentPlatformFactory(
```
Comment (tbosch) [inline modules/@angular/core/src/application_ref.ts]: This assignment is not needed.

File: modules/@angular/core/src/application_ref.ts [hunk 3]
```diff
@@ -103,12 +106,13 @@ export function createPlatformFactory(
...
             providers.concat(extraProviders).concat({provide: marker, useValue: true}));
       } else {
-        createPlatform(ReflectiveInjector.resolveAndCreate(
+        platform = createPlatform(ReflectiveInjector.resolveAndCreate(
```
Comment (tbosch) [inline modules/@angular/core/src/application_ref.ts]: This assignment is not needed.

**Review by tbosch — APPROVED** [review]
Reviewed the 2nd commit. Only minor nits .

**Pull request closed without merging**